nixos/modules/virtualisation/containers-next/container-profile.nix

```diff
@@ -0,0 +1,44 @@
+{ pkgs, lib, ... }: with lib; {
+  config = {
+    boot.isContainer = true;
+
+    # We need a static libsudoers if we bind-mount into a user-namespaced
+    # container since the bind-mounts are owned by `nouser:nogroup` then (including
+    # `/nix/store`) and this doesn't like sudo.
+    security.sudo.package = mkDefault pkgs.sudo-nspawn;
+
+    # FIXME get rid of this hack!
+    # On a test-system I experienced that this service was hanging for no reason.
+    # After a config-activation in ExecReload which affected larger parts of the OS in the
+    # container, `nixops` waited until the timeout was reached. However, the networkd
+    # was routable and the `host0` interface reached the state `configured`. Hence I'd guess
+    # that this is a networkd bug that requires investigation. Until then, I'll leave
+    # this as-is.
+    systemd.services.systemd-networkd-wait-online.serviceConfig.ExecStart = mkForce [
+      ""
+      "/run/current-system/sw/bin/true"
+    ];
+
+    # Containers are supposed to use systemd-networkd to have a proper
+    # networking stack even during boot-up.
+    networking = {
+      useHostResolvConf = false;
+      useDHCP = false;
+      useNetworkd = true;
+    };
+    systemd.network.networks."20-host0" = {
+      matchConfig = {
+        Virtualization = "container";
+        Name = "host0";
+      };
+      linkConfig.RequiredForOnline = "no";
```

Comment:
I think this line is the reason you need the hack above to disable `systemd-networkd-wait-online.service`. `host0` is effectively providing the upstream connectivity and should therefore be required for the network to be considered online, right?

Nnot sure though why `systemd-networkd-wait-online.service` just waits if there is no interface to wait for.

Reply:
> Not sure though why `systemd-networkd-wait-online.service` just waits if there is no interface to wait for.

Probably because there are some devices which don't immediately create a device (e.g. might take a while to get online, I suppose something like USB-WWAN sticks or such...

Reply:
That has worked, thanks!